extension/api.js

```diff
@@ -96,131 +283,25 @@ this.search = class extends ExtensionAPI {
       },
     });
 
-    function finish(result = false) {
-      Services.prefs.setBoolPref(RUN_ONCE_PREF, result);
-    }
-
-    // If the selected default is not the configured default for this region/locale, the
-    // user has changed default and we should not re-prompt.
-    let defaultEngine = await Services.search.getDefault();
-    let configuredDefault = await Services.search.originalDefaultEngine;
-    if (defaultEngine.name !== configuredDefault.name) {
-      console.log("reset-default-search: The default engine was already selected by the user.");
-      finish();
-      return;
-    }
+    // Ensure everything is started if this is an APP startup.
+    await searchInitialized;
+    await ExtensionSettingsStore.initialize();
 
-    // Get the latest installed addon that was installed prior to it's failure date.  Any non-app
-    // disabled state (ie. it is not disabled due to blocklisting) removes the addon from elibility.
-    let addons = (await AddonManager.getAddonsByIDs(
-        Array.from(lostEngines.keys())
-      )).filter(
-        a => a &&
-        !a.userDisabled &&
-        !a.softDisabled &&
-        !a.embedderDisabled &&
-        lostEngines.has(a.id) &&
-        a.installDate < lostEngines.get(a.id)
-      );
-    if (!addons.length) {
-      console.log("reset-default-search: No addons in our list are installed.");
-      finish();
+    await tryDefaultReset();
```

Comment:
Here this `await tryDefaultReset()` is part of `async onStartup() {`, whereas the th `tryDefaultReset` call below is not. How important is it for the logic to be blocking the `onStartup` phase? If not important, then don't make it block `onStartup`.

The relevance of this question is that any promise in `onStartup` will prevent Firefox from shutting down, as the `onStartup` method is part of extension startup (at https://searchfox.org/mozilla-central/rev/72951aa826642f048da4c6b71b8b3e36a9606dcd/toolkit/components/extensions/Extension.jsm#2658), which in itself is a shutdown blocker. If `onStartup` takes too long, then it can cause a shutdown crash like this: https://bugzilla.mozilla.org/show_bug.cgi?id=1464938